lrpar/examples/inspect/build.rs

```diff
@@ -0,0 +1,83 @@
+#![deny(rust_2018_idioms)]
+use cfgrammar::{yacc::YaccGrammar, Span};
+use lrlex::{CTLexerBuilder, DefaultLexerTypes, LRNonStreamingLexerDef, LexerDef};
+use lrpar::{LexerTypes, RTParserBuilder};
+use num_traits::ToPrimitive as _;
+use std::io::Read;
+
+pub fn set_rule_ids<LexerTypesT: LexerTypes<StorageT = u32>, LT: LexerDef<LexerTypesT>>(
+    lexerdef: &mut LT,
+    grm: &YaccGrammar,
+) -> (Option<Vec<Span>>, Option<Vec<Span>>)
+where
+    usize: num_traits::AsPrimitive<LexerTypesT::StorageT>,
+{
+    let rule_ids = grm
+        .tokens_map()
+        .iter()
+        .map(|(&n, &i)| (n, usize::from(i).to_u32().unwrap()))
+        .collect::<std::collections::HashMap<&str, u32>>();
+    let (missing_from_lexer, missing_from_parser) = lexerdef.set_rule_ids_spanned(&rule_ids);
+    let missing_from_lexer = missing_from_lexer.map(|tokens| {
+        tokens
+            .iter()
+            .map(|name| {
+                grm.token_span(*grm.tokens_map().get(name).unwrap())
+                    .expect("Given token should have a span")
+            })
+            .collect::<Vec<_>>()
+    });
+
+    let missing_from_parser =
+        missing_from_parser.map(|tokens| tokens.iter().map(|(_, span)| *span).collect::<Vec<_>>());
+    (missing_from_lexer, missing_from_parser)
+}
+
+fn main() {
+    // Since we're using both lrlex and lrpar, we use lrlex's `lrpar_config` convenience function
+    // that makes it easy to a) create a lexer and parser and b) link them together.
+    CTLexerBuilder::new()
+        .rust_edition(lrlex::RustEdition::Rust2021)
+        .lrpar_config(|ctp| {
+            ctp.rust_edition(lrpar::RustEdition::Rust2021)
+                .inspect(Box::new(move |_, recov, grm, stable, _, lexer_path| {
```

Comment:
Dumb question: are we currently thinking of this as an internal or a potentially external API? If the former, I would consider this a "test" rather than an "example".

Reply:
Because we can't `#[cfg(test)]` it easily and use it from within `cttests`, that makes it `pub`, so this is experimenting with considering it an *external* API.

Reply:
The plan to move it into a `#[test]` under *lrpar/cttests/src/lib.rs* doesn't actually work,
because while cttests lives under the `lrpar/` in the file system, from the cargo/crate perspective they
are completely separate crates, so adding it to `#[cfg(test)]` still doesn't make the callback visible within `cttests`.

Reply:
I guess I could do it within a `#[test]` under `lrpar/`

Reply:
I went ahead and reverted the addition of the example, and the making it `pub`,
and added these tests to `ctbuilder::tests` in [fcaefc9](https://github.com/softdevteam/grmtools/pull/550/commits/fcaefc969db8e932b031e8d4ae01f1cf0009e58c)

This means we aren't actually building the resulting sources, but it's not nothing?